### PR TITLE
Update page routing [Specific Instance Fix] Fix: #11737

### PR DESCRIPTION
**Pull request by Nigel2392**
Also sets the `_cached_parent_obj` on the specific instances of pages when routing.
More info can be found [here](https://github.com/wagtail/wagtail/pull/11737/files).

### CODE REVIEW

File: wagtail/models/__init__.py [hunk 1]
```diff
@@ -1678,7 +1678,9 @@ def route(self, request, path_components):
             except Page.DoesNotExist:
                 raise Http404
 
-            return subpage.specific.route(request, remaining_components)
+            specific = subpage.specific
+            setattr(specific, "_cached_parent_obj", self)
+            return specific.route(request, remaining_components)
```
Comment (zerolab) [inline wagtail/models/__init__.py:1683]: nitpic:

```suggestion
            specific_subpage = subpage.specific
            setattr(specific_subpage, "_cached_parent_obj", self)
            return specific_subpage.route(request, remaining_components)
```

**Review by zerolab — CHANGES_REQUESTED** [review]
Made a minor suggestion, but this definitely needs tests

**Comment by Nigel2392** [comment]
> Made a minor suggestion, but this definitely needs tests

Tests should have been added in the previously mentioned PR https://github.com/Nigel2392/wagtail/blob/update_page_routing/wagtail/tests/test_page_model.py#L550

I'll look to adding some more - thought I implemented it wrong at [first ](https://gist.github.com/VincentLoy/e693aa1f149a59f465a5a71b6d937b9a?permalink_comment_id=5094008#gistcomment-5094008) (thus the PR) but now I am [not so sure anymore](https://gist.github.com/VincentLoy/e693aa1f149a59f465a5a71b6d937b9a?permalink_comment_id=5094049#gistcomment-5094049).

I'll figure it out in the gist first, maybe this PR is completely redundant - seeing as the tests pass and [look correct to me](https://github.com/Nigel2392/wagtail/blob/update_page_routing/wagtail/tests/test_page_model.py#L550).

**Comment by ababic** [comment]
@Nigel2392 / @zerolab The existing code is fine. The `specific` and `specific_deferred` properties use this method under-the-hood, which copies any non-field attribute values over to the new object before returning:
https://github.com/wagtail/wagtail/blame/main/wagtail/models/specific.py#L13

Fun fact: I wrote this code originally when these methods were defined directly on the `Page` class, but that history was lost when it was bust out into it's own module

**Comment by zerolab** [comment]
@ababic that is good code, my friend. 
I did look at the specific/deferred implementation earlier today, and was thinking this is the case especially that we have a test using specific too. Had to be away from the computer so didn't follow up on this PR after the initial drive-by suggestion, so thank you for keeping an eye

**Pull request closed without merging**